### PR TITLE
Implement 7th order RadauIIA method

### PR DESCRIPTION
**Pull request by Shreyas-Ekanathan**
## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [x] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API

### CODE REVIEW

File: src/perform_step/firk_perform_step.jl [hunk 1]
```diff
@@ -749,3 +775,528 @@ end
...
+    @unpack c1, c2, c3, c4, γ, α1, β1, α2, β2, e1, e2, e3, e4, e5 = cache.tab
+    @unpack κ, cont1, cont2, cont3, cont4 = cache
+    @unpack internalnorm, abstol, reltol, adaptive = integrator.opts
+    alg = unwrap_alg(integrator, true)
+    @unpack maxiters = alg
+    mass_matrix = integrator.f.mass_matrix
+
+    # precalculations
+    rtol = @.. broadcast=false reltol^(2 / 3)/10
+    atol = @.. broadcast=false rtol*(abstol / reltol)
```
Comment (oscardssmith) [inline src/perform_step/firk_perform_step.jl:793]: where do these come from? that seems very odd...

File: src/perform_step/firk_perform_step.jl [hunk 2]
```diff
@@ -749,3 +775,528 @@ end
...
+        w4 = @.. broadcast=false TI41*z1 + TI42*z2 + TI43*z3 + TI34*z4 + TI45*z5
+        w5 = @.. broadcast=false TI51*z1 + TI52*z2 + TI53*z3 + TI54*z4 + TI55*z5
+    end
+
+    # Newton iteration
+    local ndw
+    η = max(cache.ηold, eps(eltype(integrator.opts.reltol)))^(0.8)
+    fail_convergence = true
+    iter = 0
+    while iter < maxiters
```
Comment (oscardssmith) [inline src/perform_step/firk_perform_step.jl:853]: Why are you writing out newton iterations yourself? Shouldn't this be using nlsolve like all the other implicit methods?

Comment (oscardssmith) [reply src/perform_step/firk_perform_step.jl:853]: oh, I see that all the Radau methods are like this. @ChrisRackauckas why are these using hand rolled newton solvers?

File: src/perform_step/firk_perform_step.jl (hunk 1 above)
Comment (Shreyas-Ekanathan) [reply src/perform_step/firk_perform_step.jl:793]: they're the same values used in the other Radau methods so I just copied them

File: src/perform_step/firk_perform_step.jl (hunk 2 above)
Comment (ChrisRackauckas) [reply src/perform_step/firk_perform_step.jl:853]: because it has to do a very specific FIRK newton which doesn't match the form of the others.

Comment (oscardssmith) [reply src/perform_step/firk_perform_step.jl:853]: As a general TODO, we should also move these to NonlinearSolve.jl, but that is separate from this PR.

File: src/alg_utils.jl [hunk 3]
```diff
@@ -176,7 +176,7 @@ qmin_default(alg::DP8) = 1 // 3
 qmax_default(alg::Union{OrdinaryDiffEqAlgorithm, DAEAlgorithm}) = 10
 qmax_default(alg::CompositeAlgorithm) = minimum(qmax_default.(alg.algs))
 qmax_default(alg::DP8) = 6
-qmax_default(alg::Union{RadauIIA3, RadauIIA5}) = 8
+qmax_default(alg::Union{RadauIIA3, RadauIIA5, RadauIIA7}) = 8
```
Comment (oscardssmith) [inline src/alg_utils.jl:179]: Should there be an `AbstractRadau` that these could all subclass?

**Review by oscardssmith — APPROVED** [review]

**Comment by oscardssmith** [comment]
This looks good to merge to me, (although hopefully most of this code can be deleted once we have the general order version)

**Review by ChrisRackauckas — APPROVED** [review]

File: test/algconvergence/ode_firk_tests.jl [hunk 4]
```diff
@@ -8,6 +8,11 @@ for prob in [prob_ode_linear, prob_ode_2Dlinear]
     @test sim21.𝒪est[:final]≈5 atol=testTol
 end
 
+for prob in [prob_ode_linear, prob_ode_2Dlinear]
+    sim21 = test_convergence(1 ./ 2 .^ (2.777:-1:0.777), prob, RadauIIA7())
+    @test sim21.𝒪est[:final]≈7 atol=testTol
```
Comment (ChrisRackauckas) [inline test/algconvergence/ode_firk_tests.jl:13]:
```suggestion
    @test sim21.𝒪est[:final]≈8 atol=testTol
```

**Comment by ChrisRackauckas** [comment]
Why is the IIP and OOP convergence rate so drastically different? They should give about the same answer.

**Comment by ChrisRackauckas** [comment]
oh one's 2D

File: test/algconvergence/ode_firk_tests.jl [hunk 5]
```diff
@@ -8,6 +8,12 @@ for prob in [prob_ode_linear, prob_ode_2Dlinear]
     @test sim21.𝒪est[:final]≈5 atol=testTol
 end
 
+sim21 = test_convergence(1 ./ 2 .^ (2.777:-1:0.777), prob_ode_linear, RadauIIA7())
+@test sim21.𝒪est[:final]≈8 atol=testTol
+
+sim21 = test_convergence(1 ./ 2 .^ (2.777:-1:0.777), prob_ode_2Dlinear, RadauIIA7())
+@test sim21.𝒪est[:final]≈7 atol=testTol
```
Comment (ChrisRackauckas) [inline test/algconvergence/ode_firk_tests.jl:15]:
```suggestion
@test sim21.𝒪est[:final]≈7 atol=testTol

sim21 = test_convergence(1 ./ 2 .^ (2.777:-1:0.777), prob_ode_2Dlinear, RadauIIA7())
@test sim21.𝒪est[:final]≈8 atol=testTol
```

**Comment by ChrisRackauckas** [comment]
> although hopefully most of this code can be deleted once we have the general order version

Indeed, onto the next phase.

**Pull request merged**